site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -57,20 +57,16 @@ public function openQueue() {
             );
         }
 
-        if (empty($_POST['token'])) {
-            $this->core->addErrorMessage("Missing secret code");
-            return MultiResponse::RedirectOnlyResponse(
-                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
-            );
-        }
-
         //Replace whitespace with "_"
         $queue_code = trim($_POST['code']);
-        $token = trim($_POST['token']);
+        $token = isset($_POST['token']) ? trim($_POST['token']) : '';
```

Comment:
Easier to use null coalescing operator within the `trim` to avoid the ternary.
```suggestion
        $token = trim($_POST['token'] ?? '');
```